cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -41,6 +46,7 @@
     private static final Logger log = LogManager.getLogger(StatefulSetOperator.class.getName());
     private final PodOperator podOperations;
     private final PvcOperator pvcOperations;
+    private final SecretOperator secretOperations;
```

Comment:
Needing secrets here is unfortunate, because really secrets have nothing to do with statefulset. This is supposed to be a class for handling generic statefulset, but this secret stuff is very particular to the strategi statefulsets we use for Kafka and zookeeper. The only reason having a podoperator was ok is because a statefulset is composed of pods. Can we push this logic into theKafkaAssemblyOperator?

Reply:
I could agree but the fact that the `StatefulSetOperator` class is placed in the `cluster-operator` project and not in the `operator-common` one let me think that it's not so used for "generic" statefulsets but it's more tight to K and ZK statefulsets (we have two class for them deriving from this one). 
Btw if we want to avoid using the `SecretOperator` inside this one, we should set the annotation to the `desired` before it's passed to the `reconcile`. Is that what you mean?